fastly/block_fastly_service_director.go

```diff
@@ -301,3 +318,31 @@ func flattenDirectors(directorList []*gofastly.Director, directorBackendList []*
 	}
 	return dl
 }
+
+func getDirectorBackendChange(d *schema.ResourceData, resource map[string]interface{}) (odb *schema.Set, ndb *schema.Set) {
```

Comment:
Just to double check: are we not able to reuse https://github.com/fastly/terraform-provider-fastly/blob/256c95df476fd53628927a086122178b5df0b65b/fastly/diff.go#L52 ?

Reply:
The **Diff** method cannot replace **getDirectorBackendChange()**. To my understanding, **Diff** is handling a different issue:

- The **Diff** method takes two Set objects and converts them into a DiffResult object containing the Added, Modified, Deleted, and Unmodified lists.
- **getDirectorBackendChange()** extracts child Set objects from parent Set objects and return them. The parent Sets contain old/new Directors and the child Sets contain old/new Director Backends.